runtime-modules/project-token/src/lib.rs

```diff
@@ -800,6 +801,239 @@ decl_module! {
             Self::deposit_event(RawEvent::RevenueSplitLeft(token_id, member_id, staking_info.amount));
             Ok(())
         }
+
+        /// Activate Amm functionality for the token
+        /// Preconditions
+        /// - (origin, member_id) must be a valid authentication pair
+        /// - token_id must exist
+        /// - offering state for `token_id` must be `Idle`
+        ///
+        /// Postconditions
+        /// - token `bonding_curve` activated with specified parameters
+        /// - amm account created with existential deposit (if necessary)
+        /// - event deposited
+        #[weight = 100_000_000] // TODO: adjust weight
+        fn activate_amm(origin, token_id: T::TokenId, member_id: T::MemberId, curve: BondingCurve) -> DispatchResult {
+            T::MemberOriginValidator::ensure_member_controller_account_origin(
+                origin,
+                member_id
+            )?;
+            let token_data = Self::ensure_token_exists(token_id)?;
+
+            ensure!(OfferingStateOf::<T>::ensure_idle_of::<T>(&token_data).is_ok(), Error::<T>::TokenIssuanceNotInIdleState);
```

Comment:
This is redundant, `ensure_idle_of` already returns `Error::<T>::TokenIssuanceNotInIdleState` on failure:
```suggestion
            OfferingStateOf::<T>::ensure_idle_of::<T>(&token_data)?;
```